### PR TITLE
Add example window_resize

### PR DESCRIPTION
**Pull request by Wandalen**
# Objective

I have not found any example of how to handle window resize, despite it's a problem most newbies encounters really early.

## Solution

I wrote one.
Hopefully that is somehow useful.
Let me know if any changes are required.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -257,6 +257,7 @@ Example | File | Description
 `scale_factor_override` | [`window/scale_factor_override.rs`](./window/scale_factor_override.rs) | Illustrates how to customize the default window settings
 `transparent_window` | [`window/transparent_window.rs`](./window/transparent_window.rs) | Illustrates making the window transparent and hiding the window decoration
 `window_settings` | [`window/window_settings.rs`](./window/window_settings.rs) | Demonstrates customizing default window settings
+`window_resize` | [`window/window_resize.rs`](./window/window_resize.rs) | Demonstrates how to handle window resize event
```
Comment (alice-i-cecile) [inline examples/README.md:260]:
```suggestion
`window_resize` | [`window/window_resize.rs`](./window/window_resize.rs) | Demonstrates how to handle window resize events
```

File: examples/window/window_resize.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_system(on_resize_system)
+        .add_system(exit_on_esc_system)
+        .run();
+}
+
+///
+/// The system iterates resize events and print them.
```
Comment (alice-i-cecile) [inline examples/window/window_resize.rs:16]: This should be a single `///` comment, just above the function name.

File: examples/window/window_resize.rs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+///
+
+pub fn on_resize_system(mut resize_reader: EventReader<WindowResized>) {
+    for e in resize_reader.iter() {
+        println!("event {:?}", e);
```
Comment (alice-i-cecile) [inline examples/window/window_resize.rs:21]: IMO if we want to include this example, it should do something common and nontrivial.

**Comment by Wandalen** [comment]
I would better follow principle one example, one thing, and keep samples as simple as possible. @cart @alice-i-cecile also I suggest split examples and showcase. Showcase is for marketing purposes to tiger wow effect. Examples for development. IMO it's better to stick to the single responsibility principle for development.

But I can add a camera and algorithm to fit content to the current size of the window. In CSS it is `object-fit: contain`. Like here:

image

@alice-i-cecile, do you think it is nontrivial enough?

File: examples/window/window_resize.rs (hunk 2 above)
Comment (Wandalen) [reply examples/window/window_resize.rs:15]: Fixed

**Comment by Weibye** [comment]
Hi @Wandalen, thanks for your work :) I've taken your changes and fleshed out the example even more in #5813

**Pull request closed without merging**